extensions/wikia/TemplateClassification/services/UserTemplateClassificationService.class.php

```diff
@@ -43,4 +50,23 @@ public function getType( $wikiId, $pageId ) {
 
 		return $templateType;
 	}
+
+	/**
+	 * Verify template type before user classification
+	 *
+	 * @param int $wikiId
+	 * @param int $pageId
+	 * @param string $templateType
+	 * @param string $origin
+	 * @throws BadRequestApiException
+	 */
+	public function classifyTemplate( $wikiId, $pageId, $templateType, $origin ) {
+		if ( !in_array( $templateType, self::$templateTypes ) ) {
+			throw new ApiException( self::CLASSIFY_TEMPLATE_EXCEPTION_MESSAGE, self::CLASSIFY_TEMPLATE_EXCEPTION_CODE );
+		}
+
+		parent::classifyTemplate( $wikiId, $pageId, $templateType, self::USER_PROVIDER, $origin );
+
+		wfRunHooks( 'UserTemplateClassification::TemplateClassified', [ $wikiId, $pageId, $templateType ] );
```

Comment:
In this flow, 
https://github.com/Wikia/app/pull/9206/files#diff-51080aaef54b01dd7a86c59c827a2761R124
and
https://github.com/Wikia/app/pull/9206/files#diff-2e8dea429e8ffb59895bd46f92ed5942R139
both need to build the title object.
What is the reason behind passing $pageId instead of a valid Title?